src/org/intellij/erlang/completion/QuoteInsertHandler.java

```diff
@@ -52,26 +52,39 @@ public static void process(@NotNull Project project, @NotNull String name, @NotN
 
   public static void process(@NotNull Project project, @NotNull String name, @Nullable String moduleName, @NotNull InsertionContext context) {
     if (needQuotation(project, moduleName)) {
-      Editor editor = context.getEditor();
-      Document document = editor.getDocument();
-      context.commitDocument();
       int startOffset = context.getStartOffset();
-      document.insertString(startOffset, QUOTA);
-      document.insertString(startOffset + moduleName.length() + 1, QUOTA);
+      insertQuotesIfAbsent(context, startOffset, startOffset + moduleName.length());
     }
     if (needQuotation(project, name)) {
-      Editor editor = context.getEditor();
-      Document document = editor.getDocument();
-      context.commitDocument();
       int tailOffset = context.getTailOffset();
       int startOffset = moduleName == null ? context.getStartOffset() : tailOffset - name.length();
-      document.insertString(startOffset, QUOTA);
-      document.insertString(tailOffset + 1, QUOTA);
-      editor.getCaretModel().moveToOffset(tailOffset + 2);
+      insertQuotesIfAbsent(context, startOffset, tailOffset);
+      context.getEditor().getCaretModel().moveToOffset(tailOffset + 2);
```

Comment:
I didn't like these +1's in this code, but now we also have a +2. Can't it be written properly?